stats.go

```diff
@@ -438,6 +568,44 @@ func newHistogramBucket(
 	return bucket
 }
 
+type metricExpiryTracker struct {
+	lastUpdateUnix int64
+	expired        uint32
+	scope          *scope
+	name           string
+}
+
+// expiredAndUpdated checks the metric is expired and if so
+// will return true only for the first routine that updates
+// an expired metric, signalling to insert the metric back
+// into the parent scope.
+func (m *metricExpiryTracker) expiredAndUpdated() bool {
+	// We perform a LoadUint32 first for performance reasons
+	if atomic.LoadUint32(&m.expired) == 1 && m.scope != nil &&
+		atomic.CompareAndSwapUint32(&m.expired, 1, 0) {
+		return true
+	}
+
+	return false
+}
+
+func (m *metricExpiryTracker) resetUpdatedTime() {
+	atomic.StoreInt64(&m.lastUpdateUnix, globalNow().Unix())
+}
+
+// checkExpiration returns whether the metric has expired and sets the expired
+// bit to one if true
+func (m *metricExpiryTracker) checkExpiration() bool {
+	if m.scope != nil && globalNow().Unix() >
+		atomic.LoadInt64(&m.lastUpdateUnix)+m.scope.registry.expirePeriodSeconds {
+		atomic.StoreUint32(&m.expired, 1)
```

Comment:
nit: can you put a blank line after the `{` to separate it from the multi-line condition (and can delete the one following this line)